api/types.go

```diff
@@ -1231,7 +1232,43 @@ type DeleteOrganizationMembershipPayload struct {
 }
 
 type MachineEvent struct {
-	Kind      string      `json:"kind"`
-	Timestamp time.Time   `json:"timestamp"`
-	Body      interface{} `json:"body"`
+	ID        string
+	Kind      string
+	Timestamp time.Time
+	ExitCode  *int
+	OOMKilled bool
+	// Body      map[string]interface{}
+}
+
+func (e *MachineEvent) UnmarshalJSON(data []byte) error {
+	raw := map[string]interface{}{}
+
+	if err := json.Unmarshal(data, &raw); err != nil {
+		return err
+	}
+
+	e.ID = raw["id"].(string)
+	delete(raw, "id")
```

Comment:
[nit] What's the reason for deleting data from the map?

Reply:
Nothing special, just a way of keeping only the data that has not yet been set unto a field.

Reply:
Ah ok, I think this is probably stylistic, but I don't see the point of it if the object gets garbage collected anyway once the function returns